app/views/admin/reports/_date_range_form.html.haml

```diff
@@ -1,14 +1,17 @@
 -# Field used for ransack search. This date range is mostly used for Spree::Order
 -# so default field is 'completed_at'
 - field ||= 'completed_at'
-- start_date ||= params[:q].try(:[], :completed_at_gt)
-- end_date ||= params[:q].try(:[], :completed_at_lt)
+- start_field = "#{field}_gt"
+- end_field = "#{field}_lt"
+- query = params[:q].to_h
+- start_date = datepicker_time(query[start_field].presence || 3.months.ago.beginning_of_day)
+- end_date = datepicker_time(query[end_field].presence || Time.zone.tomorrow.beginning_of_day)
```

Comment:
I wondered if we should use the current time here (`Time.zone.now`), because it's not idempotent: if you run the same query later in the day, you could get different results (eg more orders have been made).

But then it's set to when you first loaded the page, and you miss out on any new orders made since the page load.
It's more likely the user wants up to the minute results, not idempotency. So I agree with your call 👍